pytplot/QtPlotter/TVarFigure1D.py

```diff
@@ -105,11 +106,95 @@ def _visdata(self):
+                            # This triggers when we initially exceed the allowed data gap
+                            elif diff == 1 and count >= limit and not flag:
+                                count += time_accum
+                                # For some reason if you don't grab the point before the big data gap happened, when you
+                                # plot this, the data gap isn't actually removed...
+                                if consec_list[0] != 0:
+                                    consec_list.insert(0, consec_list[0]-1)
+                                # Put the current datagap into the list (overall_list) whose indices will actually be
+                                # removed from the final plot
+                                overall_list.append(consec_list)
+                                overall_list.append([nan_keys[val]])
+                                flag = True
+                            # If we already exceeded the data gap, and the current index is still part of a gap,
+                            # throw that index into the overall_list
+                            elif diff == 1 and count > limit and flag:
+                                count += time_accum
+                                overall_list.append([nan_keys[val]])
+                            # When we find that the previous index and the current one are not consecutive, stop adding
+                            # to the consec_list/overall_list (if applicable), and start over the count of time
+                            # accumulated in a gap, as well as the consecutive list of time values with nans
+                            elif diff != 1:
+                                count = 0
+                                consec_list = []
+                                flag = False
+                                # For some reason if you don't grab the point after the last point where big data gap
+                                # happened, when you plot, the data gap isn't actually removed...
+                                if nan_keys[val-1] in [y for x in overall_list for y in x]:
+                                    overall_list.append([nan_keys[val]])
+
+                    # Flatten the overall_list to just one list, instead of a list of many lists
+                    overall_list = [y for x in overall_list for y in x]
+
+                    # Remove data gaps removed based on user-input acceptable time gap
+                    # In order to do this, we set the identified indices from overall_list to 0, which in the
+                    # connect keyword argument in self.plotwindow.plot will cause that point to not be plotted
+                    time_filtered = np.array([1]*len(dataset.data.index.tolist()))
+                    for index, pos in enumerate(overall_list):
+                        # Remove unwanted time from being plotted
+                        time_filtered[pos] = 0
```

Comment:
If this is doing what I think it is doing, you can set the indexes in time_filtered without looping by using the expression

time_filtered[overall_list] = 0

Reply:
For some reason I feel like I had done just that before, and then switched for some reason... although it likely was a result of something else and i just kept the loop even though it isn't needed.